### ISSUE TITLE
Use FixedSizeBinary instead of Binary for int96 conversion when convertInt96ToArrowTimestamp is false

### ISSUE THREAD

**Issue by doki23** [apache/parquet-java#3088]
### Describe the enhancement requested

```java
public TypeMapping convertINT96(PrimitiveTypeName primitiveTypeName) throws RuntimeException {
  if (convertInt96ToArrowTimestamp) {
    return field(new ArrowType.Timestamp(TimeUnit.NANOSECOND, null));
  } else {
    return field(new ArrowType.Binary());
  }
}
```
When converting a Parquet type to an Arrow type, if the original type is int96 and the option convertInt96ToArrowTimestamp is set to false, the resulting Arrow type defaults to Binary. However, it might be more appropriate to use FixedSizeBinary instead.

**Comment by Fokko** [comment]
I agree that it would be more appropriate to return a `FixedSizeBinary`, let's ask @wgtmac, since he's the expert on Arrow. Just to make sure that there is no historical reason to return Binary instead.

**Comment by wgtmac** [comment]
Thanks for pinging me @Fokko!

I agree that `FixedSizeBinary` is more appropriate than `Binary`. However, I would argue that it is invalid to use INT96 for non-timestamp type. So I think it is better to ignore `convertInt96ToArrowTimestamp` and directly return Timestamp.
